src/lib/idb.ts

```diff
@@ -374,97 +275,90 @@ export default class IDBStorage {
-        
-        //this.log.log('IDB getAll:', fileName, request);
-      } catch(err) {
-        this.log.error('getAll error:', err, request, request.error);
-      }
+  private getObjectStore<T>(mode: IDBTransactionMode, objectStore: (objectStore: IDBObjectStore) => IDBRequest | IDBRequest[], log?: string) {
+    let perf: number;
 
-      return new Promise((resolve, reject) => {
-        const timeout = setTimeout(() => {
-          this.log.error('getAll request not finished!', request);
-          reject();
-        }, 3000);
+    if(log) {
+      perf = performance.now();
+      this.log(log + ': start');
+    }
 
-        request.onsuccess = function(event) {
-          const result = request.result;
-          if(result === undefined) {
-            reject('NO_ENTRY_FOUND');
-          } /* else if(typeof result === 'string' &&
-            result.substr(0, 5) === 'data:') {
-            resolve(dataUrlToBlob(result));
-          }  */else {
-            resolve(result);
-          }
+    return this.openDatabase().then((db) => {
+      return new Promise<T>((resolve, reject) => {
+        const transaction = db.transaction([this.storeName], mode);
 
+        transaction.onerror = (e) => {
           clearTimeout(timeout);
-        }
+          reject(transaction.error);
+        };
   
-        request.onerror = () => {
+        transaction.oncomplete = (e) => {
           clearTimeout(timeout);
-          reject();
+
+          if(log) {
+            this.log(log + ': end', performance.now() - perf);
+          }
+
+          const results = r.map(r => r.result);
+          resolve(isArray ? results : results[0]);
         };
+  
+        const timeout = setTimeout(() => {
+          this.log.error('transaction not finished', transaction);
+        }, 10000);
+  
+        /* transaction.addEventListener('abort', (e) => {
+          //handleError();
+          this.log.error('IndexedDB: transaction abort!', transaction.error);
+        }); */
+  
+        const requests = objectStore(transaction.objectStore(this.storeName));
+
+        const isArray = Array.isArray(requests);
+        const r: IDBRequest[] = isArray ? requests : [].concat(requests) as any;
+
+        // const length = r.length;
+        // /* let left = length;
+
+        // const onRequestFinished = (error?: Error) => {
+        //   if(!--left) {
+        //     resolve(result);
+        //     clearTimeout(timeout);
+        //   }
+        // }; */
+
+        // for(let i = 0; i < length; ++i) {
+        //   const request = r[i];
+        //   request.onsuccess = () => {
+        //     onRequestFinished();
+        //   };
+
+        //   request.onerror = (e) => {
+        //     onRequestFinished(transaction.error);
+        //   };
+        // }
       });
     });
   }
 
+  public getAll<T>(): Promise<T[]> {
+    return this.getObjectStore<T[]>('readonly', (objectStore) => objectStore.getAll(), DEBUG ? 'getAll' : '');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
    return this.getObjectStore<T[]>("readonly", (objectStore) => objectStore.getAll(), DEBUG ? 'getAll' : '');
```